src/mysql_k8s_helpers.py

```diff
@@ -335,16 +409,221 @@ def execute_backup_commands(
             stdout, stderr = process.wait_output()
             return (stdout, stderr)
         except ExecError as e:
-            logger.exception("Failed to execute backup script", exc_info=e)
-            logger.error(f"Stdout of script: {e.stdout}")
-            logger.error(f"Stderr of script: {e.stderr}")
+            logger.exception("Failed to execute backup commands", exc_info=e)
+            logger.error(f"Stdout: {e.stdout}")
+            logger.error(f"Stderr: {e.stderr}")
             raise MySQLExecuteBackupCommandsError(e.stderr)
         except Exception as e:
             # Catch all other exceptions to prevent the database being stuck in
             # a bad state due to pre-backup operations
-            logger.exception("Failed to execute backup script", exc_info=e)
+            logger.exception("Failed to execute backup commands", exc_info=e)
             raise MySQLExecuteBackupCommandsError(e)
 
+    def retrieve_backup_with_xbcloud(
+        self,
+        s3_bucket: str,
+        s3_path: str,
+        s3_access_key: str,
+        s3_secret_key: str,
+        backup_id: str,
+    ) -> Tuple[str, str, str]:
+        """Retrieve the specified backup from S3.
+
+        The backup is retrieved using xbcloud and stored in a temp dir in the
+        mysql container.
+        """
+        nproc_command = "nproc".split()
+        make_temp_dir_command = "mktemp --directory /tmp/mysql_sst_XXXX".split()
```

Comment:
it is important to have a temp dir on the same volume as the data-dir.
reason: `--move-back` will work much faster if both dirs are on the same volume.
it basically means the following
```suggestion
        make_temp_dir_command = "mktemp --directory /var/lib/mysql/mysql_sst_XXXX".split()
```
But keep in mind, in such case, the modifications to the `empty_data_directory` function are needed.

Reply:
addressed in 904a4f919b7f73502e4d01423c0891d85ecdc188